crates/fuel-core/src/database/block.rs

```diff
@@ -85,15 +86,10 @@ impl StorageMutate<FuelBlocks> for Database {
 }
 
 impl Database {
-    pub fn latest_height(&self) -> DatabaseResult<Option<BlockHeight>> {
-        let id = self.ids_of_latest_block()?;
-        // if no blocks, check if chain was configured with a base height
-        let id = match id {
-            Some((id, _)) => Some(id),
-            None => self.get_starting_chain_height()?,
-        };
-
-        Ok(id)
+    pub fn latest_height(&self) -> StorageResult<BlockHeight> {
+        self.ids_of_latest_block()?
+            .map(|(height, _)| height)
+            .ok_or(not_found!("BlockHeight"))
```

Comment:
I'm not too familiar, but what is the main reason for a string literal of "BlockHeight," rather than passing the `BlockHeight` type specifically?

Reply:
Yeh also should this be an option not an error?

Reply:
> I'm not too familiar, but what is the main reason for a string literal of "BlockHeight," rather than passing the BlockHeight type specifically?

Because we don't have support for not `Mappable` types=) 
<img width="787" alt="image" src="https://user-images.githubusercontent.com/18346821/213951212-316b1366-496e-4fdb-800c-88fb57b765c4.png">

> Yeh also should this be an option not an error?

`NonFound` error is an `Option`, I will describe the reason below=)